### ISSUE TITLE
BACKUP type NSR fails with missing "mminfo" utility

### ISSUE THREAD

**Issue by bruncsak** [rear/rear#2290]
* ReaR version: Relax-and-Recover 2.4 / Git

* OS version ("cat /etc/rear/os.conf"):
OS_VENDOR=RedHatEnterpriseServer
OS_VERSION=7

* ReaR configuration files ("cat /etc/rear/site.conf"):
BACKUP=NSR

* Hardware (virtual machine): Hyper-V guest

* System architecture (x86 compatible): x86_64

* Firmware (BIOS or UEFI or Open Firmware) and bootloader (GRUB or ELILO or Petitboot):
UEFI / GRUB2

* Storage: Virtual disk provided by the Hyper-V host.

* Description of the issue:

The backup method is configured to be NSR (Legato).
The "rear mkrescue" command fails with:
````
ERROR: Could not save result files with save
Aborting due to an error, check /var/log/rear/rear-<hostname>.log for details
Terminated
````
The log file contains:
````
2019-11-29 12:01:33.455795532 Including output/NSR/default/950_nsr_save_result_files.sh
2019-11-29 12:01:33.457489262 Saving result files with NSR (EMC NetWorker)
/usr/share/rear/output/NSR/default/950_nsr_save_result_files.sh: line 50: mminfo: command not found
````

Indeed, the output of the command "rpm -ql lgtoclnt | grep mminfo" is empty. The command mminfo is not shipped with Legato client package (probably any more).

**Comment by jsmeix** [comment]
@bruncsak 
regarding BACKUP=NSR:
In general there is nothing at all what I could do in case of issues
with third-party backup tools or any other third-party software
because I do not have such software on my systems
so that I can neither test nor reproduce anything.

The last adaptions and enhancements
for BACKUP=NSR were done by @hpannenb 

@hpannenb
could you again have a look here?
Of course as time permits.
Thank you in advance!

@bruncsak 
in general regarding issues for non-current ReaR versions
like what you use `Relax-and-Recover 2.4` see the section
"Testing current ReaR upstream GitHub master code" in
https://en.opensuse.org/SDB:Disaster_Recovery

**Comment by hpannenb** [comment]
@bruncsak I checked on one of our RHEL7 based servers. On these we have installed the package lgtoclnt-8.2.4.9-1 which (still) contains the mminfo command. Which version do You use? Anyway it would be interesting if there is a successor command available for the same purpose (see [http://www.ipnom.com/Legato-NetWorker-Commands/mminfo.html](url))

The initial code for the NSR workflow was included by @gdha with commit 38bb7835f08ef0e0f71ea1c2c6f0e21278572440. It might be a code change is required for newer/upcoming Legato Networker releases. My use case "NSR_CLIENT_MODE=y" does not require the mminfo.

**Comment by bruncsak** [comment]
I have the version lgtoclnt-9.2.0.5-1 installed. This version ships the following utilities:

/usr/bin/nsr_render_log
/usr/bin/nsrports
/usr/bin/recover
/usr/bin/save
/usr/sbin/dbgcommand
/usr/sbin/fsra.jar
/usr/sbin/fsra_res.jar
/usr/sbin/nsr_render_log
/usr/sbin/nsr_shutdown
/usr/sbin/nsradmin
/usr/sbin/nsrclient_nsrfa
/usr/sbin/nsrexec
/usr/sbin/nsrexecd
/usr/sbin/nsrfsra
/usr/sbin/nsrlogd
/usr/sbin/nsrlogin
/usr/sbin/nsrlogout
/usr/sbin/nsrpurge_adhoclog
/usr/sbin/nsrrcopy
/usr/sbin/nsrssltrust
/usr/sbin/nsrvbaflr
/usr/sbin/recover
/usr/sbin/save
/usr/sbin/savefs
/usr/sbin/uasm

I do not know is there any new relative to the previous version which may supersede the mminfo utility?

I tried to run the "rear mkrescue" command with "NSR_CLIENT_MODE=y" setting. It finished to run with success, I got an ISO image. I still have to test the recovery.

If the mminfo utility is unreplaceable, it would be nice to automatically switch to client mode with some reasonable defaults.

**Comment by gdha** [comment]
@bruncsak Could you ask EMC support why mminfo was removed? Perhaps, it was moved in another package?

**Comment by jsmeix** [comment]
In general regarding mandatory programs:

Mandatory programs should be listed in the REQURED_PROGS array
not in the PROGS array, cf. the comments in default.conf.

So I think in rear/conf/default.conf
```
PROGS_NSR=( nsrexec nsrexecd mminfo save savefs savepnpc recover nsrfsra nsrinfo nsrretrieve nsrwatch nsrports uasm )
```
together with prep/NSR/default/400_prep_nsr.sh
```
PROGS=( "${PROGS[@]}" "${PROGS_NSR[@]}" fmt )
```
is at least partially insufficient and should be enhanced
by an additional REQUIRED_PROGS_NSR array in default.conf like
```
PROGS_NSR=( nsrexec nsrexecd save savefs savepnpc recover nsrfsra nsrinfo nsrretrieve nsrwatch nsrports uasm )
REQUIRED_PROGS_NSR=( mminfo )
```
(add to REQUIRED_PROGS_NSR all mandatory programs for NSR)
plus in prep/NSR/default/400_prep_nsr.sh
```
PROGS+=( "${PROGS_NSR[@]}" fmt )
REQUIRED_PROGS+=( "${REQUIRED_PROGS_NSR[@]}" )
```
bcause all REQUIRED_PROGS are verified to exist in the recovery system in
https://github.com/rear/rear/blob/master/usr/share/rear/build/default/990_verify_rootfs.sh#L204

**Comment by bruncsak** [comment]
> @bruncsak Could you ask EMC support why mminfo was removed? Perhaps, it was moved in another package?

@gdha, I checked with EMC support. The support said that the extended client package (lgtoxclnt) contains the mminfo binary. However, I did not install this immediately, rather upgraded first the lgtoclnt package to match the version of the server. With lgtoclnt-19.1.1.1-1.x86_64 the extended client is lgtoxtdclnt-19.1.1.1-1.x86_64 (slightly different prefix).

The mminfo binary utility is available now.
